From 8acbcaa6b3d855f3e85e6a108db323278584091e Mon Sep 17 00:00:00 2001
From: Loic Akue <loicx.akue@intel.com>
Date: Mon, 3 Nov 2014 16:26:42 +0100
Subject: [PATCH] bug readline

|			   READLINE PATCH REPORT
|			   =====================
|
|Readline-Release: 6.3
|Patch-ID: readline63-002
|
|Bug-Reported-by:	Anatol Pomozov <anatol.pomozov@gmail.com>
|Bug-Reference-ID:	<CAOMFOmXy3mT2So5GQ5F-smCVArQuAeBwZ2QKzgCtMeXJoDeYOQ@mail.gmail.com>
|Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-readline/2014-03/msg00010.html
|
|Bug-Description:
|
|When in callback mode, some readline commands can cause readline to seg
|fault by passing invalid contexts to callback functions.

Signed-off-by: Loic Akue <loicx.akue@intel.com>

diff --git a/readline.c b/readline.c
index 03eefa6..55c0522 100644
--- a/readline.c
+++ b/readline.c
@@ -744,7 +744,8 @@ _rl_dispatch_callback (cxt)
     r = _rl_subseq_result (r, cxt->oldmap, cxt->okey, (cxt->flags & KSEQ_SUBSEQ));

   RL_CHECK_SIGNALS ();
-  if (r == 0)			/* success! */
+  /* We only treat values < 0 specially to simulate recursion. */
+  if (r >= 0 || (r == -1 && (cxt->flags & KSEQ_SUBSEQ) == 0))	/* success! or failure! */
     {
       _rl_keyseq_chain_dispose ();
       RL_UNSETSTATE (RL_STATE_MULTIKEY);
--
1.7.9.5

